allennlp/models/encoder_decoders/wikitables_semantic_parser.py

```diff
@@ -271,6 +310,166 @@ def forward(self,  # type: ignore
             # TODO(matt): compute accuracy here.
             return outputs
 
+    @staticmethod
+    def _get_neighbor_indexes(worlds: List[WikiTablesWorld],
+                              num_entities: int,
+                              tensor: torch.Tensor) -> Variable:
+        """
+        This method returns the indices of each entity's neighbors. A tensor
+        is accepted as a parameter for copying purposes
+
+        Parameters
+        ----------
+        worlds : ``List[WikiTablesWorld]``
+        num_entities : ``int``
+        tensor : ``torch.Tensor``
+            Used for copying the constructed list onto the right device.
+
+        Returns
+        -------
+        A ``torch.autograd.Variable`` with shape ``(batch_size, num_entities, num_entities)``.
+        Dimension 2 contains the indices for the neighbors of the entity represented by the
+        index of dimension 1. Dimension 2 is limited to num_entities as it is assumed that
+        each entity can have a number of neighbors at most equal to the total number of entities.
+        """
+        batches_neighbors = []
+        for world in worlds:
+            # Each batch instance has its own world, which has a corresponding table.
+            entities = world.table_graph.entities
+            entity2indices = {entity: i for i, entity in enumerate(entities)}
```

Comment:
I'm not a huge fan of using `2` like this, but if you like it, that's fine.  I think `entity_indices` and `entity_neighbors` are a little easier to read, though I can see an argument that if you're consistent, maybe the `2` is a strong signal that this is a dict.  If you want to keep it with the `2` though, this should be `entity2index`, not `entity2indices`, because you're mapping to an `int`, not a `List[int]`.

Reply:
And I guess your use of `entity_neighbors` below is another decent argument for giving this a separate name that makes it obvious it's a dict...